addons/pos_self_order/static/src/app/self_order_service.js

```diff
@@ -280,6 +283,60 @@ export class SelfOrder extends Reactive {
         );
 
         this.currentCategory = this.pos_category.length > 0 ? [...this.categoryList][0] : null;
+
+        if (this.kitchen_printers) {
+            for (const printerConfig of Object.values(this.kitchen_printers)) {
+                const printer = this.create_printer(printerConfig);
+                if (printer) {
+                    printer.config = printerConfig;
+                    this.kitchenPrinters.push(printer);
+                }
+            }
+        }
+    }
+
+    create_printer(printer) {
+        const url = deduceUrl(printer.proxy_ip || "");
+        return new HWPrinter({ url });
+    }
+
+    _getPrintingCategoriesChanges(categories) {
+        return this.currentOrder.lines.filter((orderline) =>
+            categories.some((categId) =>
+                this.productByIds[orderline["product_id"]].pos_categ_ids
+                    .map((categ) => categ.id)
+                    .includes(categId)
+            )
+        );
+    }
+
+    async printChanges() {
+        const d = new Date();
+        let hours = "" + d.getHours();
+        hours = hours.length < 2 ? "0" + hours : hours;
+        let minutes = "" + d.getMinutes();
+        minutes = minutes.length < 2 ? "0" + minutes : minutes;
+        for (const printer of this.kitchenPrinters) {
+            const orderlines = this._getPrintingCategoriesChanges(
+                Object.values(printer.config.product_categories_ids)
+            );
+            if (orderlines) {
+                const printingChanges = {
+                    new: orderlines,
+                    tracker: this.currentOrder.table_stand_number,
+                    trackingNumber: this.currentOrder.trackingNumber || "unknown number",
+                    name: this.currentOrder.pos_reference || "unknown order",
+                    time: {
+                        hours,
+                        minutes,
+                    },
+                };
```

Comment:
May I know why aren't we reusing the "changes" method from pos? Is it because for self-ordering, user can always append items? They can't remove?